gdal/frmts/saga/sagadataset.cpp

```diff
@@ -406,16 +406,45 @@ GDALDataset *SAGADataset::Open( GDALOpenInfo * poOpenInfo )
 
 {
     /* -------------------------------------------------------------------- */
-    /*  We assume the user is pointing to the binary (i.e. .sdat) file.     */
+    /*  We assume the user is pointing to the binary (i.e. .sdat) file or a */
+    /*  compressed raster (.sg-grd-z) file.                                 */
     /* -------------------------------------------------------------------- */
-    if( !EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sdat"))
+    if( !EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sdat") &&
+        !EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sg-grd-z") )
     {
         return NULL;
     }
 
-    const CPLString osPath = CPLGetPath( poOpenInfo->pszFilename );
-    const CPLString osName = CPLGetBasename( poOpenInfo->pszFilename );
-    const CPLString osHDRFilename = CPLFormCIFilename( osPath, osName, ".sgrd" );
+    
+    CPLString osPath, osName, osHDRFilename; if (EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sg-grd-z"))
+    {
+        const char* fileName = CPLGetFilename(poOpenInfo->pszFilename);
+        CPLString osPath("/vsizip/{");
+        osPath += fileName;
+        osPath += "}/";
+
+        char ** filesinzip = VSIReadDir(osPath);
```

Comment:
The result of VSIReadDir() must be free'd with CSLDestroy() once you're done with it